### DISCUSSION TITLE
Override by filename doesn't work

### DISCUSSION THREAD

**Discussion by hrkcz001** [kyazdani42/nvim-web-devicons#279] [Q&A]
What am I doing wrong?
```
local icon, color = require('nvim-web-devicons').get_icon_color("hxx")
require('nvim-web-devicons').setup({
    strict = true,
    override_by_filename = {
        [".*/include/.*"] = {
            icon = icon,
            color = color,
            name = "Header"
        }
    }
})
```
![image](https://github.com/nvim-tree/nvim-web-devicons/assets/37705573/5e8f3dcf-daf4-4689-a00d-947ff42c18d4)

I've tried the setup example code chunk(.gitignore) for filenames like "element", but it doesn't work either.

**Comment by alex-courtis** [comment]
Regexes are not supported, only exact file name matches.

**Reply by alex-courtis** [reply]
> I've tried the setup example code chunk(.gitignore) for filenames like "element", but it doesn't work either.

Please post your call to setup so that we may try to replicate.

**Reply by hrkcz001** [reply]
> > I've tried the setup example code chunk(.gitignore) for filenames like "element", but it doesn't work either.
> 
> Please post your call to setup so that we may try to replicate.

```
lazy.setup({
...
{
    'nvim-tree/nvim-web-devicons',
    config = function()
        local icon, color = require('nvim-web-devicons').get_icon_color("hxx")
        require('nvim-web-devicons').setup({
            strict = true,
            override_by_filename = {
                ["element"] = {
                    icon = icon,
                    color = color,
                    name = "Header"
                }
            }
        })
    end
} ,                                                                              
...
{
    'nvim-tree/nvim-tree.lua',
    dependencies = {
        'nvim-tree/nvim-web-devicons'
    },
    config = function()
        require('nvim-tree').setup({ 
            filters = { dotfiles = true }
        })
    end
},
...                                           
})                                                  
```

![image](https://github.com/nvim-tree/nvim-web-devicons/assets/37705573/993ff099-697c-4422-8f92-d72bd2da071c)

**Reply by hrkcz001** [reply]
> Regexes are not supported, only exact file name matches.

And there is no built-in option to do that?

**Reply by alex-courtis** [reply]
> And there is no built-in option to do that?

Searching the large number of icon types for regexes would be very slow. Applications very frequently get the icons and would likely need to cache them.
